easier_net/sier_net.py

```diff
@@ -46,21 +48,26 @@ def __init__(
         self.num_classes = num_classes
         assert num_classes != 1
         self.weight = weight
+
+        self.max_iters = max_iters
+        self.max_prox_iters = max_prox_iters
         self.criterion = (
             nn.CrossEntropyLoss(weight=torch.Tensor(weight))
             if self.num_classes >= 2
             else nn.MSELoss()
         )
+        
         self.score_criterion = (
             nn.CrossEntropyLoss() if self.num_classes >= 2 else nn.MSELoss()
         )
-        self.net = SierNet(
-            input_filter_layer=self.input_filter_layer,
-            n_layers=self.n_layers,
-            n_input=self.n_inputs,
-            n_hidden=self.n_hidden,
-            n_out=self.n_out,
-        )
+
+
+        if state_dict is not None:
+            self.net.load_state_dict(state_dict, strict=False)
```

Comment:
this will crash, because you dont have a neural network yet, i.e. you haven't defined `self.net` yet. I think we have to move line 248 back here? or is there a way to just load the state dictionary directly without figuring out the number of output nodes?

Reply:
ooh, I think we moved the self.net here back when we were trying to resolve the torch.Size issue, but I might just need to move this back and resolve the size issue some other way